lib/msal-common/src/client/OnBehalfOfClient.ts

```diff
@@ -60,6 +60,12 @@ export class OnBehalfOfClient extends BaseClient {
         const cachedAccessToken = this.readAccessTokenFromCache();
         if (!cachedAccessToken ||
             TimeUtils.isTokenExpired(cachedAccessToken.expiresOn, this.config.systemOptions.tokenRenewalOffsetSeconds)) {
+
+            // Update the server telemetry outcome
+            !cachedAccessToken ?
+                this.serverTelemetryManager?.setCacheOutcome(CacheOutcome.NO_CACHED_ACCESS_TOKEN) :
+                this.serverTelemetryManager?.setCacheOutcome(CacheOutcome.CACHED_ACCESS_TOKEN_EXPIRED);
```

Comment:
If you aren't using the returned value, can you use an if statement instead? A little more readable:
```suggestion
            if (!cachedAccessToken) {
                this.serverTelemetryManager?.setCacheOutcome(CacheOutcome.NO_CACHED_ACCESS_TOKEN) :
            } else {
                this.serverTelemetryManager?.setCacheOutcome(CacheOutcome.CACHED_ACCESS_TOKEN_EXPIRED);
            }
```

Reply:
Agreed, both implementations do add to the readability of this code, I have accepted the second version.